src/ImageIO/CGImageDestination.cs

```diff
@@ -470,5 +495,17 @@ public bool CopyImageSource (CGImageSource image, CGImageDestinationOptions opti
 					o.Dispose ();
 			}
 		}
+
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[DllImport (Constants.ImageIOLibrary)]
+		static extern void CGImageDestinationAddAuxiliaryDataInfo (IntPtr /* CGImageDestinationRef* */ idst, IntPtr /* CFStringRef* */ auxiliaryImageDataType, IntPtr /* CFDictionaryRef* */ auxiliaryDataInfoDictionary);
+
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		public void AddAuxiliaryDataInfo (CGImageDestination dest, CGImageAuxiliaryDataType auxiliaryImageDataType, CGImageAuxiliaryDataInfo auxiliaryDataInfo)
+		{
+			using (var dict = auxiliaryDataInfo?.ToDictionary ()) {
+				CGImageDestinationAddAuxiliaryDataInfo (dest.GetHandle (), auxiliaryImageDataType.GetConstant ().GetHandle (), dict.GetHandle ());
```

Comment:
GetHandle is an extension method, so it won't crash when dict is null.  For a null value, dict.GetHandle () should return IntPtr.Zero